### PR TITLE
[editorial] Fix doc page titles and add Hugo front matter

### LINKED ISSUE #169: Consistent page title form and capitalization
This is what the side nav looks like for the semconv pages, as (soon to be) published on the OTel website:

> image

~(1) I'd like to suggest that all section titles (at this level and recurisively too) be of the form:~

- ~Xxx Semantic Conventions~

~instead of "Semantic Conventions for Yyy".~

(2) There should be consistency in capitalization. I'm ok with whichever convention you use:Title Case or Sentence case -- the latter being the Google dev docs recommendation).

(3) Some titles are "Semantic Convention", singular. This should be fixed.

WDYT @jsuereth @joaopgrassi

/cc @svrnm @cartermp

### PR DESCRIPTION
**Pull request by chalin**
- Closes #169
- Closes #170 because it supersedes it (either, really, can be merged first)
- Adds the necessary Hugo front matter to be able to publish pages on the OTel website

/cc @svrnm @cartermp @jsuereth @joaopgrassi

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: Semantic Conventions
+# no_list: true
+cascade:
+  body_class: otel-docs-spec
+  github_repo: &repo https://github.com/open-telemetry/semantic-conventions
+  github_subdir: docs
+  path_base_for_github_subdir: content/en/docs/specs/semconv/
```
Comment (jsuereth) [inline docs/README.md:8]: To some extent, these `path_base` selections are actually implementation details of the docs workflow.

Should this specific value be something inserted by the adjust pages perl script?


I'm worried about keeping path_base up-to-date as we add new markdown files. 

At a minimum, there should be a section under CONTRIBUTING.md about how to fill it out and why it's needed.

At a maximum, I think perhaps the `path_base` or `from` sections should be automatically derived on the docs side.  I'm happy to contribute that.

File: docs/faas/faas-metrics.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <!--- Hugo front matter used to generate the website version of this page:
-linkTitle: FaaS
+linkTitle: Metrics
```
Comment (jsuereth) [inline docs/faas/faas-metrics.md:2]: Should this be FaaS Metrics ?

**Review by jsuereth — COMMENTED** [review]
Thanks for doing all this work!

I have a meta-question related to how we keep this hugo matter up to date, given the pace at which we've been adding docs.

TL;DR; we either need a guide on how/what to fill out and when *or* for some of this can we move it into the docs `adjust-pages.pl` script?

File: docs/faas/faas-metrics.md (hunk 2 above)
Comment (chalin) [reply docs/faas/faas-metrics.md:2]: The given title results in this side nav:

> image

It seems redundant to include "FaaS" in a title under the "FaaS" section, but it's your call.

Comment (jsuereth) [reply docs/faas/faas-metrics.md:2]: Ah, got it.  SO contextually, the title is always under the previous title.  LGTM

**Comment by chalin** [comment]
@jsuereth - you make good points about some of the front matter. To answer your questions:

- One of my goals is to **eliminate** the `adjust-pages.pl` script, and replace it by a _very simple and generic_ Gulp task that only activates the front matter and little else.
- I agree that embedding the `path_base_for_github_subdir` isn't ideal. Yes, that could be handled by the Gulp task.

Would it be ok for this PR to land, and we address your concerns (regarding `path_base_for_github_subdir` etc) iteratively?

**Comment by chalin** [comment]
I've rebased and re-normalized the new and updated doc pages.

**Comment by jsuereth** [comment]
@chalin 

> Would it be ok for this PR to land, and we address your concerns (regarding path_base_for_github_subdir etc) iteratively?

Yes, depending on the answer to this question:

If we add new markdown files that do NOT include appropriate hugo frontmatter/github redirect, does it fundamentally break the website?

If the answer is YES, I think we should add a `CONTRIBUTING.md` section requiring hugo front matter of `linkTitle`.

**Comment by chalin** [comment]
@jsuereth - no, not having front matter for new files **will not break** the website build, and the page will get published. The `linkTitle` field is mainly to define a shorter title name that gets used in the left-side nav and breadcrumbs -- so it's a "nice to have". For example, with the link titles added by this PR, the left sidenav now looks like the same as the list of pages shown in the docs landing page. It's nice to have that correspondence.

**Review by jsuereth — APPROVED** [review]
Great!  Thanks for all the fixes, this looks good to go.

File: docs/README.md (hunk 1 above)
Comment (jsuereth) [reply docs/README.md:8]: Discussed below.  Since this is a non-breaking, we can accept PR as-is.

We can flesh out CONTRIBUTING.md later.

**Review by reyang — APPROVED** [review]
